### DISCUSSION TITLE
Can Pinecil Read USB-C Cable E-Marker Chips?

### DISCUSSION THREAD

**Discussion by Garywoo** [Ralim/IronOS#1905] [Q&A] [answered]
I'd like to verify the wattage rating of some USB-C cables according to their [E-Marker chips](https://www.totalphase.com/blog/2020/10/what-is-e-marker-how-does-it-work/). Can the Pinecil display this information somehow?

I looked at the [PD debug menu](https://ralim.github.io/IronOS/DebugMenu/#pd-debug-menu), but I'm not sure if I can interpret E-Marker data from this.

**Comment by Ralim** [comment, marked as answer]
Theoretically it could but doesn't as of present.

This would need to be implemented in the USB-pd library and exposed somehow on the device.

**Reply by Nickduino** [reply]
A testing menu that displays the E-Marker chip of the cables **and** the compatible voltages and current of the power supply sounds like a cool "debug idea"

**Reply by Ralim** [reply]
Yeah at the moment it only does the charger advertised values. 

What emarker data would people like to see?

**Reply by Garywoo** [reply]
I've opened a new enhancement issue with details about what E-Marker data I'd like to see. #1914
